### ISSUE TITLE
Linux Dedicated Server Crash, Looking For Missing Directory

### ISSUE THREAD

**Issue by Peter65535** [Regalis11/Barotrauma#8863]
- [x] I have searched the issue tracker to check if the issue has already been reported.

**Description**
After the latest update, the linux dedicated server is crashing on start due to a missing folder, instead of creating said folder.

**Steps To Reproduce**
- Be running a linux dedicated server on the previous version (0.16.X? Whatever was the latest stable before today's update)
- Update the server with SteamCMD (automatically done by LinuxGSM in my case)
- Server crashes on start

**Version**
Linux dedicated server, Game version 0.17.12.0 (ReleaseLinux, branch release, revision 2971300777)
With LinuxGSM v21.5.1

**Additional information**
This can, of course, be worked around by manually creating the folder the server is looking for. In my case that was `/home/berriesbarotrauma/.local/share/Daedalic Entertainment GmbH/Barotrauma/WorkshopMods/Installed`.

The latest crash report is below, the other 7 are identical except for time, as the monitoring attempted to restart the server when it saw that it was not running.

```
berriesbarotrauma@games:~/serverfiles$ cat "servercrashreport (7).log"
﻿Barotrauma Dedicated Server crash report (generated on 4/20/2022 6:30:42 PM)


Barotrauma seems to have crashed. Sorry for the inconvenience!


Game version 0.17.12.0 (ReleaseLinux, branch release, revision 2971300777)
Language: English
Selected content packages: None
Level seed: no level loaded
Loaded submarine: None
Selected screen: None


System info:
    Operating system: Unix 5.4.0.105 64 bit


Exception: Could not find a part of the path '/home/berriesbarotrauma/.local/share/Daedalic Entertainment GmbH/Barotrauma/WorkshopMods/Installed'. (System.IO.DirectoryNotFoundException)
Target site: IntPtr CreateDirectoryHandle(System.String, Boolean)
Stack trace:
   at System.IO.Enumeration.FileSystemEnumerator`1.CreateDirectoryHandle(String path, Boolean ignoreNotFound)
   at System.IO.Enumeration.FileSystemEnumerator`1.Init()
   at System.IO.Enumeration.FileSystemEnumerator`1..ctor(String directory, Boolean isNormalized, EnumerationOptions options)
   at System.IO.Enumeration.FileSystemEnumerable`1..ctor(String directory, FindTransform transform, EnumerationOptions options, Boolean isNormalized)
   at System.IO.Enumeration.FileSystemEnumerableFactory.UserDirectories(String directory, String expression, EnumerationOptions options)
   at System.IO.Directory.InternalEnumeratePaths(String path, String searchPattern, SearchTarget searchTarget, EnumerationOptions options)
   at Barotrauma.Steam.SteamManager.Workshop.DeleteFailedCopies() in <DEV>/Barotrauma/BarotraumaShared/SharedSource/Steam/Workshop.cs:line 253
   at Barotrauma.ContentPackageManager.Init()+MoveNext() in <DEV>/Barotrauma/BarotraumaShared/SharedSource/ContentManagement/ContentPackageManager.cs:line 444
   at Barotrauma.Extensions.IEnumerableExtensions.Consume[T](IEnumerable`1 enumerable) in <DEV>/Barotrauma/BarotraumaShared/SharedSource/Extensions/IEnumerableExtensions.cs:line 149
   at Barotrauma.GameMain.Init() in <DEV>/Barotrauma/BarotraumaServer/ServerSource/GameMain.cs:line 109
   at Barotrauma.GameMain.Run() in <DEV>/Barotrauma/BarotraumaServer/ServerSource/GameMain.cs:line 290
   at Barotrauma.Program.Main(String[] args) in <DEV>/Barotrauma/BarotraumaServer/ServerSource/Program.cs:line 75


Last debug messages:
```

**Comment by esirprus** [comment]
i got the same problem on my two linux  Dedicated Server.
`﻿Barotrauma Dedicated Server crash report (generated on 2022-4-21 上午10:21:15)


Barotrauma seems to have crashed. Sorry for the inconvenience! 


Game version 0.17.12.0 (ReleaseLinux, branch release, revision 2971300777)
Language: Simplified Chinese
Selected content packages: None
Level seed: no level loaded
Loaded submarine: None
Selected screen: None


System info:
    Operating system: Unix 3.10.0.1160 64 bit


Exception: Could not find a part of the path '/home/btserver/.local/share/Daedalic Entertainment GmbH/Barotrauma/WorkshopMods/Installed'. (System.IO.DirectoryNotFoundException)
Target site: IntPtr CreateDirectoryHandle(System.String, Boolean)
Stack trace: 
   at System.IO.Enumeration.FileSystemEnumerator`1.CreateDirectoryHandle(String path, Boolean ignoreNotFound)
   at System.IO.Enumeration.FileSystemEnumerator`1.Init()
   at System.IO.Enumeration.FileSystemEnumerator`1..ctor(String directory, Boolean isNormalized, EnumerationOptions options)
   at System.IO.Enumeration.FileSystemEnumerable`1..ctor(String directory, FindTransform transform, EnumerationOptions options, Boolean isNormalized)
   at System.IO.Enumeration.FileSystemEnumerableFactory.UserDirectories(String directory, String expression, EnumerationOptions options)
   at System.IO.Directory.InternalEnumeratePaths(String path, String searchPattern, SearchTarget searchTarget, EnumerationOptions options)
   at Barotrauma.Steam.SteamManager.Workshop.DeleteFailedCopies() in <DEV>/Barotrauma/BarotraumaShared/SharedSource/Steam/Workshop.cs:line 253
   at Barotrauma.ContentPackageManager.Init()+MoveNext() in <DEV>/Barotrauma/BarotraumaShared/SharedSource/ContentManagement/ContentPackageManager.cs:line 444
   at Barotrauma.Extensions.IEnumerableExtensions.Consume[T](IEnumerable`1 enumerable) in <DEV>/Barotrauma/BarotraumaShared/SharedSource/Extensions/IEnumerableExtensions.cs:line 149
   at Barotrauma.GameMain.Init() in <DEV>/Barotrauma/BarotraumaServer/ServerSource/GameMain.cs:line 109
   at Barotrauma.GameMain.Run() in <DEV>/Barotrauma/BarotraumaServer/ServerSource/GameMain.cs:line 290
   at Barotrauma.Program.Main(String[] args) in <DEV>/Barotrauma/BarotraumaServer/ServerSource/Program.cs:line 75


Last debug messages:
`
another crash report
`﻿Barotrauma Dedicated Server crash report (generated on 2022/4/21 上午10:05:20)


Barotrauma seems to have crashed. Sorry for the inconvenience! 


Game version 0.17.12.0 (ReleaseLinux, branch release, revision 2971300777)
Language: Simplified Chinese
Selected content packages: None
Level seed: no level loaded
Loaded submarine: None
Selected screen: None


System info:
    Operating system: Unix 5.4.0.92 64 bit


Exception: Could not find a part of the path '/home/btserver/.local/share/Daedalic Entertainment GmbH/Barotrauma/WorkshopMods/Installed'. (System.IO.DirectoryNotFoundException)
Target site: IntPtr CreateDirectoryHandle(System.String, Boolean)
Stack trace: 
   at System.IO.Enumeration.FileSystemEnumerator`1.CreateDirectoryHandle(String path, Boolean ignoreNotFound)
   at System.IO.Enumeration.FileSystemEnumerator`1.Init()
   at System.IO.Enumeration.FileSystemEnumerator`1..ctor(String directory, Boolean isNormalized, EnumerationOptions options)
   at System.IO.Enumeration.FileSystemEnumerable`1..ctor(String directory, FindTransform transform, EnumerationOptions options, Boolean isNormalized)
   at System.IO.Enumeration.FileSystemEnumerableFactory.UserDirectories(String directory, String expression, EnumerationOptions options)
   at System.IO.Directory.InternalEnumeratePaths(String path, String searchPattern, SearchTarget searchTarget, EnumerationOptions options)
   at Barotrauma.Steam.SteamManager.Workshop.DeleteFailedCopies() in <DEV>/Barotrauma/BarotraumaShared/SharedSource/Steam/Workshop.cs:line 253
   at Barotrauma.ContentPackageManager.Init()+MoveNext() in <DEV>/Barotrauma/BarotraumaShared/SharedSource/ContentManagement/ContentPackageManager.cs:line 444
   at Barotrauma.Extensions.IEnumerableExtensions.Consume[T](IEnumerable`1 enumerable) in <DEV>/Barotrauma/BarotraumaShared/SharedSource/Extensions/IEnumerableExtensions.cs:line 149
   at Barotrauma.GameMain.Init() in <DEV>/Barotrauma/BarotraumaServer/ServerSource/GameMain.cs:line 109
   at Barotrauma.GameMain.Run() in <DEV>/Barotrauma/BarotraumaServer/ServerSource/GameMain.cs:line 290
   at Barotrauma.Program.Main(String[] args) in <DEV>/Barotrauma/BarotraumaServer/ServerSource/Program.cs:line 75


Last debug messages:
`

**Comment by Regalis11** [comment]
Fixed in https://github.com/Regalis11/Barotrauma-development/commit/3c1d04213ad0757575c32e49edbc4ff18bb67d65

**Comment by theCrius** [comment]
Same problem here. Is this going to be released soon? I can't access the fixed link (private I guess?)

**Comment by theCrius** [comment]
I want to add that previous version of common dockerfiles where looking for the mods folder in dedicated server in the INSTALL_LOCATION/mods.

As an example, this command look for the default `mods` folder and move it into an external virtual volume, linking it back
```
RUN mv $INSTALL_LOC/Mods $MODS_LOC && \
    ln -s $MODS_LOC $INSTALL_LOC/Mods
```

In case this has been changed, it would be nice to know :)

**Comment by theCrius** [comment]
Thanks for the reply @Regalis11, I've done just that but also noticed that by default there is no `config_player.xml` which seems to be required anyway so I extrapolated one from my local config, getting rid what seems to be the client-side part (audio, graphic, etc) and it seems to work.

I've made a small repo with a working docker image for it given that I've had the configuration done anyway.
I'll try and keep it updated for when this small bugs are fixed but anyone feel free to fork or make pull requests as needed.
https://github.com/theCrius/barotrauma-docker/

Cheers
